### DISCUSSION TITLE
Ho to change video quality on youtube videos?

### DISCUSSION THREAD

**Discussion by elec2** [dyc3/opentogethertube#636] [Q&A] [answered]
I can't find any settings for that because i don't see the original youtube controls. Is the video supposed to change the quality automatically? Because even that is not working on my end.

**Comment by dyc3** [comment, marked as answer]
In short, you can't, and there's not much that I can do about it.

You used to be able to, but that functionality was removed from the Youtube iframe API. It's _supposed_ to adjust automatically, but in my experience it doesn't work that well. It might be possible if we completely abandon the iframe player, but that could cause maintenance issues.

**Reply by elec2** [reply]
Ah ok, hm but why it is working on sync-tube? 
[see here](https://sync-tube.de/rooms/ckyY8YvAYJ)

**Reply by dyc3** [reply]
Because they are using Youtube's built in controls. We hide those in favor of using our own set of video controls instead. We intentionally don't show Youtube's controls because its harder to keep the video in sync.

**Reply by elec2** [reply]
Seems legit but if we can't really use it at all because of 480p potato qualli?!
The sync on sync-tube is not really that bad. I think 1 second max.

**Reply by dyc3** [reply]
IIRC, there is a workaround.

If you click the video title when you mouse over the video, it will open the video in a new tab. Then you can change the quality/turn on captions. Then if you refresh OTT, it should keep the settings.

But yeah, I'm extremely frustrated that Youtube removed the APIs to change the video quality. Maybe we could make it so that its possible to show Youtube's controls temporarily.

**Reply by elec2** [reply]
hm the workaround is not so good in my case. Well, I have to deal with people who don't even know how to spell "internet" correctly. 
But I think the temporary controls should be a good solution for that.

**Reply by hofq** [reply]
Was there ever a fix or a workaround implemented?

**Reply by dyc3** [reply]
@hofq The only workaround that exists I have already mentioned in this thread. https://github.com/dyc3/opentogethertube/discussions/636#discussioncomment-2212057
